src/Conversion/KrnlToLLVM/KrnlEntryPoint.cpp

```diff
@@ -84,6 +84,29 @@ class KrnlEntryPointOpLowering : public OpRewritePattern<KrnlEntryPointOp> {
         createEntryBlock(dynEntryPointFuncTy, dynamicEntryPointFunc, loc);
     rewriter.setInsertionPointToStart(&entryPointEntryBlock);
 
+    // Emit code to initialize accelerators by calling OMInitCompatibleAccelX
+    // where X is the accelerator name.
+    // OMInitCompatibleAccelX's signature is `void (i64)`.
+    if (Attribute maccelAttr =
+            module->getAttrOfType<::mlir::Attribute>("onnx-mlir.accels")) {
+      assert(
+          maccelAttr.isa<ArrayAttr>() && "onnx-mlir.accels must be ArrayAttr");
+      ArrayAttr accels = maccelAttr.cast<ArrayAttr>();
+      for (uint64_t i = 0; i < accels.size(); ++i) {
+        assert(accels[i].isa<StringAttr>() && "Attribute must be StringAttr");
+        StringRef accelStr = accels.getValue()[i].cast<StringAttr>().getValue();
+        std::pair<StringRef, StringRef> NameAndVersion = accelStr.split('-');
+        uint64_t versionNumberInHex =
+            std::stoul(NameAndVersion.second.str(), nullptr, 16);
+        FlatSymbolRefAttr funcRef = getOrInsertOMInitCompatibleAccel(
+            rewriter, module, NameAndVersion.first);
+        LLVM::ConstantOp versionNumberVal = rewriter.create<LLVM::ConstantOp>(
+            loc, int64Ty, rewriter.getI64IntegerAttr(versionNumberInHex));
```

Comment:
The parameter is actually unsigned, but it probably does not cause any issue here.